### ISSUE TITLE
Improve polynomial fitting

### ISSUE THREAD

**Issue by mykelk** [pjabardo/CurveFit.jl#3]
I see that you have:

``` julia
least_squares{T<:Number}(A::StridedVecOrMat{T}, y::Vector{T}) = A \ y
```

I think you want to replace the `A\y` with something like this:

``` julia
 (q, r) = qr(A)
 r \ (q' * y)
```

The different is typically only apparent when you go to high-degree polynomials. You can see it here:
![image](https://cloud.githubusercontent.com/assets/6294079/4125541/c9fa85e2-32e2-11e4-98a4-2837e637b8c0.png)
Your method is shown in blue, but the qr method that I'm suggesting is in red. Things become even more apparent when we go to 8th order:
![image](https://cloud.githubusercontent.com/assets/6294079/4125549/daf19d7c-32e2-11e4-8914-69898608dee2.png)

This is the data I used for the plot:

``` julia
srand(0)
s = 1:10
u = 4 - 0.5 * s + 0.35 * s.^2 - 0.03 * s.^3 + rand(length(s))
```

**Comment by ianshmean** [comment]
To back this up, I'm getting very poor poly fitting with a different dataset

![image](https://cloud.githubusercontent.com/assets/1694067/23866584/a352f50c-07ef-11e7-836b-887ec88b98ea.png)

```
df = readtable("data.csv", header = false)
x = collect(df[1])
y0 = collect(df[2])

figure(figsize=(12.5,5))
for n in 1:10
    fit = curve_fit(Poly, x, y0,n)
    y0b = fit(x) 
    subplot(2,5,n)
    plot(x,y0,"bo",x,y0b,"r-", linewidth=1)
    title(string("n = ",n))
end
```

**Comment by pjabardo** [comment]
I have been solving all the issues with the package and I noticed this old issue. Sorry for missing it for so long.

I couldn't reproduce the example by @mykelk . On the other hand, the examples given by @ianshmean are reproducible. The interesting thing is that If I divide x by 1e10, the curve fitting appears to be doing fine. I will try to find out what is happening and try to solve the problem.

Paulo

**Comment by pjabardo** [comment]
I solved the problem and released a new version 0.3.0
